addons/sale/sale.py

```diff
@@ -117,6 +120,7 @@ def _compute_tax_id(self):
     pricelist_id = fields.Many2one('product.pricelist', string='Pricelist', required=True, readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, help="Pricelist for current sales order.")
     currency_id = fields.Many2one("res.currency", related='pricelist_id.currency_id', string="Currency", readonly=True, required=True)
     project_id = fields.Many2one('account.analytic.account', 'Analytic Account', readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}, help="The analytic account related to a sales order.", copy=False)
+    related_project_id = fields.Many2one('account.analytic.account', inverse='_inverse_project_id', related='project_id', string='Analytic Account', help="The analytic account related to a sales order.", copy=False)
```

Comment:
I guess the point of this is to be able to write on the fields when it's not set but then lock it in place. Am I right?

Quite ingenious, but I'm not a fan of hacks (this could stop working for a number of reasons, related fields on the model itself is really).

So while I do tip my hat for ingenuity, I'll try to think of something else (I'll check with @rco-odoo when I have the occasion, he might have a better way to do this).